### PR TITLE
[v6] Add project overview and contributing documentation

### PR DESCRIPTION
**Pull request by joheredi**
Adding documentation for project design overview and contributing information

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+<a id="org38012c9"></a>
+
+## Repository
+
+-   Autorest.Typescript is being developed in the Azure/Autorest.Typescript repository
+-   The version under development is known as v6 and the working branch is "v6", master contains the olver v5 version.
+-   In the new version the package name has changed to @autorest/typescript
+
+
+<a id="org422658b"></a>
```
Comment (chradek) [inline docs/contributing.md]: Your org mode is showing :)

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+# Autorest.Typescript Dev Workflow
+
+
+<a id="org38012c9"></a>
+
+## Repository
+
+-   Autorest.Typescript is being developed in the Azure/Autorest.Typescript repository
+-   The version under development is known as v6 and the working branch is "v6", master contains the olver v5 version.
```
Comment (chradek) [inline docs/contributing.md]:
```suggestion
-   The version under development is known as v6 and the working branch is "v6", master contains the older v5 version.
```

File: docs/overview.md [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+### What is Autorest.Typescript
+
+- Autorest.Typescript is a plugin for Autorest (core) which takes the code model generated buy Autorest (core) and generates a Typescript SDK
+- Autorest.Typescript is written in Typescript (of course)
+
+<a id="org47f1d3e"></a>
+
+### Why writing a new version of Autorest
+
+- The main reason is that, to support OpenAPI v4, changes in the previous version would be substantial requiring big architecture changes, so a re-write was a more attractive investment
```
Comment (chradek) [inline docs/overview.md]:
```suggestion
- The main reason is that, to support OpenAPI v3, changes in the previous version would be substantial requiring big architecture changes, so a re-write was a more attractive investment
```

We're adding support for version 3 right?

**Comment by ramya-rao-a** [comment]
@deyaaeldeen,
Can you review this PR and help @joheredi land it?

**Comment by deyaaeldeen** [comment]
@ramya-rao-a roger than, I am on it.

File: docs/contributing.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    
+        npm run test
+
+-   Run Unit Tests only
+    Unit Test mostly tests units in the Transforms layer
+
+    npm run unit-test
+
+-   Run Integration Tests
+    Integration tests run against the Test Server, running this command will do the following things:
```
Comment (deyaaeldeen) [inline docs/contributing.md:60]: you did not specify the command here.

**Comment by joheredi** [comment]
Thanks for helping with this @deyaaeldeen! It is great to have you look at this with fresh eyes! Please feel free to update anything you feel not-accurate,

File: docs/contributing.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+### Code Generation
+
+To debug code generation the flag &#x2013;typescript.debugger can be used
+
+1.  Start Debugging Code Generation of test scenarios
+
+    -   You can debug the generation of a test scenario by appending the flag to the generation command for example:
+        
+            npm run generate-bodycomplex -- --typescript.debugger
```
Comment (deyaaeldeen) [inline docs/contributing.md:95]: I do not see the `generate-bodycomplex` script in `package.json`.

File: docs/contributing.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+
+<a id="org9754a77"></a>
+
+# Autorest.Typescript Dev Workflow
+
+
+<a id="org38012c9"></a>
+
+## Repository
```
Comment (deyaaeldeen) [inline docs/contributing.md:24]: Perhaps outlining the directory structure of the v6 would be helpful :)

File: docs/contributing.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+-   Clone the repository[ Azure/Autorest.Typescript](https://github.com/Azure/autorest.typescript.git)
+-   Switch to branch v6
+    
+        git fetch upstream v6
+        git checkout v6
+
+-   Install Autorest Core
+    
+        npm install -g @autorest/autorest
```
Comment (deyaaeldeen) [inline docs/contributing.md:43]: I see `"autorest": "^3.0.6187"` in the deps in `package.json`, is this still required? should `"autorest` be a peer dependency instead?

File: docs/contributing.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+### Before investing more time investigating
+
+-   Make sure to reset autorest, this will ensure that the versions used by Auotrest Core are the correct ones. This will solve many problems
+    
+        autorest-beta --reset
+
+
+<a id="orge680821"></a>
+
+### I've made changes to Autorest.Typescript, however when re-generating the library my changes doesn't seem to have been picked up
```
Comment (deyaaeldeen) [inline docs/contributing.md:164]:
```suggestion
### I've made changes to Autorest.Typescript, however when re-generating the library my changes don't seem to have been picked up
```

File: docs/overview.md [hunk 9]
```diff
@@ -0,0 +1,457 @@
...
+<a id="org89a3069"></a>
+## Overview
+
+<a id="org89a3066"></a>
+
+### What is Autorest (core)
+
+- Autorest is an OpenAPI (Swagger) specification code generator.
+- The new version of autorest (v3) supports [OpenApi v3](https://github.com/OAI/OpenAPI-Specification)
+- Autorest v3 uses[ @azure/modelerfour](https://github.com/Azure/autorest.modelerfour) to parse the OpenAPI (Swagger) document and generates a language agnostic code model
```
Comment (deyaaeldeen) [inline docs/overview.md:31]:
```suggestion
- Autorest v3 uses[ @azure/modelerfour](https://github.com/Azure/autorest.modelerfour) to parse the OpenAPI (Swagger) document and generates a language agnostic code model.
```

File: docs/overview.md [hunk 10]
```diff
@@ -0,0 +1,457 @@
...
+
+<a id="org89a3069"></a>
+## Overview
+
+<a id="org89a3066"></a>
+
+### What is Autorest (core)
+
+- Autorest is an OpenAPI (Swagger) specification code generator.
+- The new version of autorest (v3) supports [OpenApi v3](https://github.com/OAI/OpenAPI-Specification)
```
Comment (deyaaeldeen) [inline docs/overview.md:30]:
```suggestion
- The new version of autorest (v3) supports [OpenApi v3](https://github.com/OAI/OpenAPI-Specification).
```

File: docs/contributing.md (hunk 7 above)
Comment (joheredi) [reply docs/contributing.md:43]: This is recommended to be installed globally when you need to test code-generation outside of the Autorest.Typescript repo.

For example, say that the Azure Tables team report an issue with the generated output. You make the changes locally in Autorest.Typescript and want to test those changes in the azure-sdk-for-js repo. For this you will need to run `autorest --typescript --use=<Path to autorest.typescript repo>`

**Pull request merged**